megamek/src/megamek/common/util/fileUtils/ScaledImageFileFactory.java

```diff
@@ -0,0 +1,135 @@
+
+    //region Getters/Setters
+    public int getWidth() {
+        return width;
+    }
+
+    public void setWidth(int width) {
+        this.width = width;
+    }
+
+    public int getHeight() {
+        return height;
+    }
+
+    public void setHeight(int height) {
+        this.height = height;
+    }
+    //endregion Getters/Setters
+
+    /**
+     * Get the <code>ItemFile</code> for the given <code>File</code>.
+     *
+     * @param file The input <code>File</code> object that will be read to produce the item.
+     *             This value must not be <code>null</code>.
+     * @return an <code>ItemFile</code> for the given file.
+     */
+    @Override
+    public ItemFile getItemFile(final File file) {
+        // Validate the input.
+        Objects.requireNonNull(file, "A null image file was passed.");
+
+        // Construct an anonymous class that gets an Image for the file.
+        return new ItemFile() {
+            private File itemFile = file; // copy the file entry
+            private Image image = null; // cache the Image
+
+            @Override
+            public Object getItem() {
+                // Cache the image on first use.
+                if (image == null) {
+                    String name = itemFile.getAbsolutePath();
+                    image = ImageUtil.loadImageFromFile(name);
+                }
+                // Return a copy of the image.
+                return ImageUtil.getScaledImage(image, getWidth(), getHeight());
+            }
+        };
+    }
+
+    /**
+     * Get the <code>ItemFile</code> for the given <code>ZipEntry</code> in the <code>ZipFile</code>.
+     *
+     * @param zipEntry The <code>ZipEntry</code> that will be read to produce the item. This value
+     *                 must not be <code>null</code>.
+     * @param zipFile The <code>ZipFile</code> object that contains the <code>ZipEntry</code>
+     *                that will produce the item. This value must not be <code>null</code>.
+     * @return an <code>ItemFile</code> for the given zip file entry.
+     */
+    @Override
+    public ItemFile getItemFile(final ZipEntry zipEntry, final ZipFile zipFile) {
+        // Validate the input.
+        Objects.requireNonNull(zipEntry, "A null ZIP entry was passed.");
+        Objects.requireNonNull(zipFile, "A null ZIP file was passed.");
+
+        // Construct an anonymous class that gets an Image for the file.
+        return new ItemFile() {
+            private ZipEntry itemEntry = zipEntry; // copy the ZipEntry
+            private Image image = null; // cache the Image
+
+            @Override
+            public Object getItem() throws Exception {
+                // Cache the image on first use.
+                if (image == null) {
+                    image = createZippedImage(itemEntry, zipFile);
+                }
+                // Return a copy of the image.
+                return ImageUtil.getScaledImage(image, getWidth(), getHeight());
+            }
```

Comment:
Should we be caching the scaled image as long as height/width don't change? Or do we scale them a lot?

Reply:
Pre-existing... but good point. Will swap that over

Reply:
It has to do with cloning the image. I'll check and see if that's required anywhere else.

Reply:
From testing it does not appear to be required, and thus I've pushed that change.